### PR TITLE
new: aws-c-common package

### PR DESCRIPTION
**Pull request by leodido**
Version 0.6.9

### CODE REVIEW

File: packages/a/aws-c-common/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        -- Build type
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+
+        -- Configs
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (shared and "ON" or "OFF"))
+        table.insert(configs, "-DPERFORM_HEADER_CHECK=ON")
+        table.insert(configs, "-DENABLE_NET_TESTS=OFF")
+
+        import("package.tools.cmake").install(package, configs)
+    end)
```
Comment (waruqi) [inline packages/a/aws-c-common/xmake.lua:26]: please add on_test

File: packages/a/aws-c-common/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    end
+
+    on_install("linux", function (package)
+        local shared = package:config("shared")
+        local configs = {}
+
+        -- Build type
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+
+        -- Configs
```
Comment (waruqi) [inline packages/a/aws-c-common/xmake.lua:20]: remove comments

File: packages/a/aws-c-common/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        add_syslinks("dl", "m", "pthread", "rt")
+    end
+
+    on_install("linux", function (package)
+        local shared = package:config("shared")
+        local configs = {}
+
+        -- Build type
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+
```
Comment (waruqi) [inline packages/a/aws-c-common/xmake.lua:19]: remove empty line

File: packages/a/aws-c-common/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        -- Configs
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (shared and "ON" or "OFF"))
```
Comment (waruqi) [inline packages/a/aws-c-common/xmake.lua:21]: package:config("shared") and "ON" or "OFF"

File: packages/a/aws-c-common/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    if is_plat("linux") then
+        add_syslinks("dl", "m", "pthread", "rt")
+    end
+
+    on_install("linux", function (package)
+        local shared = package:config("shared")
+        local configs = {}
+
+        -- Build type
```
Comment (waruqi) [inline packages/a/aws-c-common/xmake.lua:17]: remove comments

File: packages/a/aws-c-common/xmake.lua (hunk 4 above)
Comment (leodido) [reply packages/a/aws-c-common/xmake.lua:21]: I'm using the `shared` local variable declared above. Isn't that good? Am I missing something?

Comment (waruqi) [reply packages/a/aws-c-common/xmake.lua:20]: If it is only used in this line of code, there is no need to define additional variables.

**Pull request merged**